### ISSUE TITLE
GraphiQL issue

### ISSUE THREAD

**Issue by sandervanhooft** [Folkloreatelier/laravel-graphql#65]
Hi,

I've just tried the GraphiQL support. It seems to work,  a bit. I can query my GraphQL schema through it.

But unfortunately:

- Docs pane does not load contents
- Autocomplete does not work
- I get an error in red in the right results pane on startup (I tried running gulp):

```
invariant@https://unpkg.com/graphiql@0.7.8/graphiql.min.js:11:7912
defineFieldMap@https://unpkg.com/graphiql@0.7.8/graphiql.min.js:12:9851
getFields@https://unpkg.com/graphiql@0.7.8/graphiql.min.js:12:14838
https://unpkg.com/graphiql@0.7.8/graphiql.min.js:13:12924
typeMapReducer@https://unpkg.com/graphiql@0.7.8/graphiql.min.js:13:13087
reduce@[native code]
e@https://unpkg.com/graphiql@0.7.8/graphiql.min.js:13:16026
buildClientSchema@https://unpkg.com/graphiql@0.7.8/graphiql.min.js:14:5879
https://unpkg.com/graphiql@0.7.8/graphiql.min.js:1:29756
promiseReactionJob@[native code]
```

Also, I noticed that the Docs pane does not load, and I cannot use autocomplete. So something seems to be going wrong here. Any clue?

The browser console shows no errors though.

Kind regards,

Sander

**Comment by dmongeau** [comment]
Concerning the error in the right pane, do you have any query or mutation defined in your config? Because I had this error too but it was just because there was nothing defined in my schema. GraphiQL doesn't seems to like that.

**Comment by dmongeau** [comment]
Definitely seems the problem. After I've added Query and Mutation, everything works.

<img width="874" alt="capture d ecran 2016-11-14 08 50 53" src="https://cloud.githubusercontent.com/assets/124966/20267182/8a326770-aa47-11e6-9aea-85004306efcc.png">

**Comment by sandervanhooft** [comment]
I have a query and type defined, no mutation. They seem to be bug free (I received a different error when they weren't ;) )

Perhaps you have an idea where to start debugging this (as my console shows nothing)?

**Comment by dmongeau** [comment]
I will need to check but I was getting this error when I had no query OR no mutation. You need at least one of both. I'll check if it's normal with GraphiQL or if it's a bug from this package.

**Comment by dmongeau** [comment]
Is this the error you are getting? If yes, it kind of says what is the bug. There is no mutation like I said in the previous comment.

![capture d ecran 2016-11-14 14 20 52](https://cloud.githubusercontent.com/assets/124966/20279204/9539f2fe-aa75-11e6-8100-c3e20a395042.png)

**Comment by pelletiermaxime** [comment]
I remember googling this error and finding this https://github.com/nuwave/laravel-graphql-relay/issues/1. My API is read only so I had to create an empty mutation for graphiql to work :/

**Comment by dmongeau** [comment]
Yes, this solved the problem for me!

**Comment by sandervanhooft** [comment]
@pelletiermaxime @dmongeau Adding an empty mutation indeed solved the issue, thanks!

**Comment by petecoop** [comment]
Fixed this here #88
